tests/functional/glusterd/test_validate_optimized_glusterd_handshake_for_2000_volumes.py

```diff
@@ -0,0 +1,192 @@
+#  Copyright (C) 2021 Red Hat, Inc. <http://www.redhat.com>
+#
+#  This program is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 2 of the License, or
+#  any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License along
+#  with this program; if not, write to the Free Software Foundation, Inc.,
+#  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+""" Description:
+      TC to validate optimized handshake by glusterd with 2000 volumes
+"""
+
+from random import choice
+from time import sleep
+from glusto.core import Glusto as g
+from glustolibs.gluster.exceptions import ExecutionError
+from glustolibs.gluster.gluster_base_class import GlusterBaseClass
+from glustolibs.gluster.volume_ops import get_volume_list
+from glustolibs.gluster.gluster_init import (stop_glusterd, restart_glusterd,
+                                             wait_for_glusterd_to_start)
+from glustolibs.gluster.volume_libs import (cleanup_volume)
+from glustolibs.gluster.volume_ops import (set_volume_options,
+                                           get_volume_options,
+                                           volume_create,
+                                           volume_start)
+from glustolibs.gluster.peer_ops import (is_peer_connected,
+                                         peer_probe_servers,
+                                         peer_detach,
+                                         nodes_from_pool_list)
+from glustolibs.gluster.lib_utils import get_servers_bricks_dict
+
+
+class TestValidateOptimizedGlusterdHandshake(GlusterBaseClass):
+
+    def setUp(self):
+        """ Detach peers, and leave it as a 3 node cluster """
+        for server in self.servers[3:]:
+            ret, _, _ = peer_detach(self.mnode, server)
+            if ret != 0:
+                raise ExecutionError("Peer detach failed")
+
+        self.get_super_method(self, 'setUp')()
+
+    def tearDown(self):
+        """ Cleanup the volumes """
+        if self.glusterd_is_stopped:
+            ret = restart_glusterd(self.servers[1])
+            self.assertTrue(ret, "Failed to start glusterd on node: %s"
```

Comment:
Please refrain from using assert statements in tearDown use ExecutionError instead.